### PR TITLE
[FIX] move and improve gap alphabet

### PR DESCRIPTION
**Pull request by eldariont**

### CODE REVIEW

File: include/seqan3/alphabet/gap/gap.hpp [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    /* types */
+    using char_type = char;
+    using integral_type = uint8_t;
+    using c_type = bool;
+
+    /* member */
+    c_type value;
+
+    /* static member */
+    static constexpr char_type gap_symbol{'-'};
```
Comment (h-2) [inline include/seqan3/alphabet/gap/gap.hpp]: you can remove this and just return it from the function...

Comment (h-2) [inline include/seqan3/alphabet/gap/gap.hpp]: you can remove this and just return it from the function...

File: include/seqan3/alphabet/gap/gap.hpp [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+#include "../alphabet.hpp"
+
+namespace seqan3
+{
+
+struct gap
+{
+    /* types */
+    using char_type = char;
+    using integral_type = uint8_t;
```
Comment (h-2) [inline include/seqan3/alphabet/gap/gap.hpp]: integral type could be bool, too, right?

File: include/seqan3/alphabet/gap/gap.hpp [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    using c_type = bool;
+
+    /* member */
+    c_type value;
```
Comment (h-2) [inline include/seqan3/alphabet/gap/gap.hpp]: this could be constexpr and be fixed to 0

File: include/seqan3/alphabet/gap/gap.hpp [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    {
+        value = 0;
+        return *this;
+    }
+
+    constexpr gap from_integral(integral_type const in)
+    {
+        value = static_cast<c_type>(in);
+        return *this;
+    }
```
Comment (h-2) [inline include/seqan3/alphabet/gap/gap.hpp]: the from_ functions could be no-ops, right?

File: include/seqan3/alphabet/gap/gap.hpp [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
```
Comment (h-2) [inline include/seqan3/alphabet/gap/gap.hpp]: remove empty line

File: include/seqan3/alphabet/gap/gap.hpp [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+// Author: Hannes Hauswedell <hannes.hauswedell@fu-berlin.de>
+// Author: Marcel Ehrhardt <marcel.ehrhardt@fu-berlin.de>
+// Author: David Heller <david.heller@fu-berlin.de>
+// ==========================================================================
+
+#pragma once
+
+#include <optional>
+
+#include "../alphabet.hpp"
```
Comment (rrahn) [inline include/seqan3/alphabet/gap/gap.hpp]: should we really use relative paths. 
I find it is always easier to read where the stuff is located, if it is written as absolute path?
Just a question!

**Pull request closed without merging**